docs/examples.rst

```diff
@@ -0,0 +1,179 @@
+    >>> # Run before every test for reproducibility
+    >>> def seed_all():
+    ...     np.random.seed(123)
+    ...     tf.random.set_seed(123)
+    >>>
+
+.. testsetup:: *
+
+    import numpy as np
+    import tensorflow as tf
+    import deepchem as dc
+
+    # Run before every test for reproducibility
+    def seed_all():
+        np.random.seed(123)
+        tf.random.set_seed(123)
+
+
+SAMPL (FreeSolv)
+----------------
+
+Examples of training models on the SAMPL(FreeSolv) dataset included in `MoleculeNet <./moleculenet.html>`_.
+
+We'll be using its :code:`smiles` field to train models to predict its experimentally measured solvation energy (:code:`expt`).
+
+MultitaskRegressor
+^^^^^^^^^^^^^^^^^^
+
+First, we'll load the dataset with :func:`load_sampl() <deepchem.molnet.load_sampl>` and fit a :class:`MultitaskRegressor <deepchem.models.MultitaskRegressor>`:
+
+.. doctest:: sampl
+
+    >>> seed_all()
+    >>> # Load SAMPL dataset with default 'index' splitting
+    >>> SAMPL_tasks, SAMPL_datasets, transformers = dc.molnet.load_sampl()
+    >>> SAMPL_tasks
+    ['expt']
+    >>> train_dataset, valid_dataset, test_dataset = SAMPL_datasets
+    >>>
+    >>> # We want to know the pearson R squared score, averaged across tasks
+    >>> avg_pearson_r2 = dc.metrics.Metric(dc.metrics.pearson_r2_score, np.mean)
+    >>> 
+    >>> # We'll train a multitask regressor (fully connected network)
+    >>> model = dc.models.MultitaskRegressor(
+    ...     len(SAMPL_tasks),
+    ...     n_features = 1024,
+    ...     layer_sizes=[1000],
+    ...     dropouts=[.25],
+    ...     learning_rate=0.001,
+    ...     batch_size=50)
+    >>> 
+    >>> # Fit trained model (returns average loss over the most recent checkpoint interval)
+    >>> model.fit(train_dataset)
+    0.1726440668106079
+    >>> 
+    >>> # We now evaluate our fitted model on our training and validation sets
+    >>> model.evaluate(train_dataset, [avg_pearson_r2], transformers)
+    {'mean-pearson_r2_score': 0.9244964295814636}
+    >>> model.evaluate(valid_dataset, [avg_pearson_r2], transformers)
+    {'mean-pearson_r2_score': 0.7532658569385681}
+
+
+GraphConvModel
+^^^^^^^^^^^^^^
+The default `featurizer <./featurizers.html>`_ for SAMPL is :code:`ECFP`, short for
+`"Extended-connectivity fingerprints." <./featurizers.html#circularfingerprint>`_
+For a :class:`GraphConvModel <deepchem.models.GraphConvModel>`, we'll reload our datasets with :code:`featurizer='GraphConv'`:
+
+.. doctest:: sampl
+
+    >>> seed_all()
+    >>> # Load SAMPL dataset
+    >>> SAMPL_tasks, SAMPL_datasets, transformers = dc.molnet.load_sampl(
+    ...     featurizer='GraphConv')
+    >>> train_dataset, valid_dataset, test_dataset = SAMPL_datasets
+    >>>
+    >>> model = dc.models.GraphConvModel(len(SAMPL_tasks), mode='regression')
+    >>> 
+    >>> # Fit trained model (returns average loss over the most recent checkpoint interval)
+    >>> model.fit(train_dataset, nb_epoch=20)
+    0.05753047466278076
```

Comment:
Having an explicit number here might cause a doctest failure (since even on successful runs there will be some variance). Perhaps we should add a directive to doctest to not test output of this line?

Reply:
Good to know setting `tf.random.set_seed(123)` stably reproduces the same result! There might still be issues when TF undergoes version upgrades and the like so I think setting `#doctest:+SKIP` might make for more robust infrastructure